src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1836,7 +1836,7 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
             // transform raw results into entities
             const rawRelationIdResults = await relationIdLoader.load(rawResults);
             const rawRelationCountResults = await relationCountLoader.load(rawResults);
-            const transformer = new RawSqlResultsToEntityTransformer(this.expressionMap, this.connection.driver, rawRelationIdResults, rawRelationCountResults, this.queryRunner);
+            const transformer = new RawSqlResultsToEntityTransformer(this.expressionMap, this.connection, rawRelationIdResults, rawRelationCountResults, this.queryRunner);
```

Comment:
I don't like that the whole connection should be transferred to transformer.
I like `Single responsibility principles`

Reply:
Do you have any suggestions?